### DISCUSSION TITLE
What should I use instead of 'on event' which will be deprecated

### DISCUSSION THREAD

**Discussion by Michael-indochat** [statelyai/xstate#4154] [Q&A] [answered]
How should this example use other methods instead of on event ?

![image](https://github.com/statelyai/xstate/assets/114893149/fb2cebf9-e013-420a-91c2-8fa9e4cd0734)

**Comment by davidkpiano** [comment, marked as answer]
You should type this as `always: /* transition */`

```diff
-on: { '': 'three' }
+always: 'three'
```

Or better yet, use a transition object (preferred):
```ts
// on: { '': 'three' }
// always: 'three'
always: { target: 'three' }
```

**Reply by Michael-indochat** [reply]
thanks a lot!
